api/envoy/extensions/certificate_selectors/on_demand_secret/v3/config.proto

```diff
@@ -0,0 +1,38 @@
+syntax = "proto3";
+
+package envoy.extensions.certificate_selectors.on_demand_secret.v3;
+
+import "envoy/config/core/v3/config_source.proto";
+
+import "udpa/annotations/status.proto";
+import "validate/validate.proto";
+
+option java_package = "io.envoyproxy.envoy.extensions.certificate_selectors.on_demand_secret.v3";
+option java_outer_classname = "ConfigProto";
+option java_multiple_files = true;
+option go_package = "github.com/envoyproxy/go-control-plane/envoy/extensions/certificate_selectors/on_demand_secret/v3;on_demand_secretv3";
+option (udpa.annotations.file_status).package_version_status = ACTIVE;
+
+// [#protodoc-title: On-demand secret certificate selector]
+// [#extension: envoy.certificate_selectors.on_demand_secret]
+
+// Fetches the secret on-demand while allowing the parent cluster or listener to accept connections
+// without warming. The certificate secret is validated against the outer common TLS context, e.g.
+// by setting FIPS compliance policy on the loaded secret material.
+message Config {
+  message SdsSecretConfig {
+    // Secret name. If specified, the secret is pre-fetched during configuration and applied to all connections.
+    string name = 1;
+
+    // If specified, the secret is fetched and reloaded via SDS. When absent, the secret is loaded from the
+    // static resources.
+    config.core.v3.ConfigSource sds_config = 2;
+  }
+
+  // Defines the source of the secrets (static resources or dynamic SDS).
+  SdsSecretConfig sds_secret_config = 1 [(validate.rules).message = {required: true}];
+
+  // If specified and there is a filter state with the given key, then the object string value determines the secret name.
+  string filter_state_key = 2;
```

Comment:
I think from an API view, this doesn't need to be configurable; it should advertise (via documentation) that setting a filter state with a certain name will cause the secret name to be overridden.